tasks/main.yml

```diff
@@ -9,59 +9,89 @@
     k8s_auth_api_key: "{{ ansible_env.K8S_AUTH_API_KEY }}"
   when: ansible_env.K8S_AUTH_API_KEY is defined
 
-- name: Create/update namespace
-  k8s:
-    context: "{{ k8s_kube_context }}"
-    kubeconfig: "{{ k8s_kube_config }}"
-    definition: "{{ lookup('template', 'namespace.yaml.j2') }}"
-    state: present
-    # Ensure we see any failures in CI
-    wait: yes
-    validate:
-      fail_on_error: yes
-      strict: yes
-  when: not k8s_auth_api_key
+# Does the user need to set k8s_auth_host?
+- when: not k8s_auth_host
+  fail:
+    msg: "k8s_auth_host must be set. Please see the role documentation."
 
-- name: Create/update deploy account
-  k8s:
-    context: "{{ k8s_kube_context }}"
-    kubeconfig: "{{ k8s_kube_config }}"
-    definition: "{{ lookup('template', 'deploy_account.yaml.j2') }}"
-    state: present
-    # Ensure we see any failures in CI
-    wait: yes
-    validate:
-      fail_on_error: yes
-      strict: yes
-  register: deploy_account
-  when: not k8s_auth_api_key
+# FROM here on, always specify
+#
+#          host: "{{ k8s_auth_host }}"
+#
+# when communicating with the cluster. This makes sure we're talking to the right one.
 
-- name: Get deploy account token
-  k8s_info:
-    kind: Secret
-    namespace: "{{ k8s_namespace }}"
-    name: "{{ deploy_account.result.results.0.result.secrets.0.name }}"
-  register: deploy_account_token
-  when: not k8s_auth_api_key
+# Does the user need to set k8s_auth_ssl_ca_cert?
+- when: not k8s_auth_ssl_ca_cert
+  block:
+    - command: kubectl config view --minify=true -o jsonpath='{.clusters[0].cluster.certificate-authority-data}' --raw
+      register: data_command
+    - debug:
+         msg: |
+           1) Create a file (maybe k8s_auth_ssl_ca_cert.txt) and set contents to
+           {{ data_command.stdout | b64decode }}
+           with all indentation removed. (This does not need to be secret.)
+           2) Set k8s_auth_ssl_ca_cert to the path to the file.
+           3) Run again after doing that
 
-- name: Set k8s_auth_api_key
-  set_fact:
-    k8s_auth_api_key: "{{ deploy_account_token.resources.0.data.token | b64decode }}"
-  when: not k8s_auth_api_key
+# Does the user need to set k8s_auth_api_key?
+- when: not k8s_auth_api_key
+  block:
```

Comment:
I kept running into a hard to parse ``SSLError`` stack trace here and wondered if adding tasks here with a helpful error message would be useful to other users:

```yaml
block:
  - name: Query nodes to check cluster connection
    k8s_info:
      kind: Node
      host: "{{ k8s_auth_host }}"
    register: node_query_result
    ignore_errors: True

  - name: Fail play if above Node query was unsuccessful
    fail:
      msg: "Unable to access the cluster. Are you using the correct kubectl context?"
    when:
      - "'rc' in node_query_result"
      - node_query_result.rc == 1
```